### PR TITLE
Physics tests: kinematics

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context
New circleCI tests for physics functions introduced in PR #91  

CI download test object dataset

Test initialization of default PhysicsManager and related code path for "enable_physics" flag.

Test object loading, instancing, removal, kinematics functions, etc...

## How Has This Been Tested

pytest circleCI

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Testing

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: tests/test_physics.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+import numpy as np
+import pytest
+
+import examples.settings
+import habitat_sim
+
+
+# return True if val_1 is within epsilon error of val_2
+def within_eps_error(val_1, val_2):
```
Comment (erikwijmans) [inline tests/test_physics.py]: Unused function?

File: tests/test_physics.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    # test get and set transform
+    sim.set_transformation(I, object_id)
+    assert np.allclose(sim.get_transformation(object_id), I)
+
+    # test get and set rotation
+    Q = habitat_sim.utils.quat_from_angle_axis(np.pi, np.array([0, 1.0, 0]))
+    sim.set_rotation(habitat_sim.utils.quat_to_magnum(Q), object_id)
+    assert np.allclose(
+        sim.get_transformation(object_id),
+        np.array(
```
Comment (erikwijmans) [inline tests/test_physics.py]: Maybe construct this via 

```
expected = np.eye(4)
expected[0:3, 0:3] = quaternion.as_rotation_matrix(Q)
```

(you'll need an `import quaternion` at the top)

File: tests/test_physics.py (hunk 1 above)
Comment (aclegg3) [reply tests/test_physics.py]: Yeah, ended up using np.allclose(), haha.

File: tests/test_physics.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    # test get and set rotation
+    Q = habitat_sim.utils.quat_from_angle_axis(np.pi, np.array([0, 1.0, 0]))
+    expected = np.eye(4)
+    expected[0:3, 0:3] = quaternion.as_rotation_matrix(Q)
+    sim.set_rotation(habitat_sim.utils.quat_to_magnum(Q), object_id)
+    assert np.allclose(sim.get_transformation(object_id), expected)
+    assert np.allclose(
+        habitat_sim.utils.quat_from_magnum(sim.get_rotation(object_id)),
+        habitat_sim.utils.quat_from_coeffs(np.array([0.0, 1.0, 0.0, 0.0])),
```
Comment (erikwijmans) [inline tests/test_physics.py]:
```suggestion
        Q,
```
isn't this just `Q`?

**Comment by aclegg3** [comment]
Any more thoughts on this one? Ready to merge.

File: src/esp/gfx/Simulator.h [hunk 4]
```diff
@@ -88,7 +88,7 @@ class Simulator {
   int getPhysicsObjectLibrarySize();
 
   // remove object objectID instance in sceneID
-  void removeObject(const int objectID, const int sceneID = 0);
+  int removeObject(const int objectID, const int sceneID = 0);
```
Comment (mosra) [inline src/esp/gfx/Simulator.h]: If I get it correctly, [my comment](https://github.com/facebookresearch/habitat-sim/pull/213#discussion_r325322324) from #213 on the return type change should have been on this PR instead? :)

Comment (aclegg3) [reply src/esp/gfx/Simulator.h]: The comments should be up to date now here.

**Pull request merged**